site_scons/site_tools/doneapi.py

```diff
@@ -0,0 +1,83 @@
+"""SCons.Tool.doneapi
+
+Hack to support oneapi version of Intel compilers
+
+"""
+import sys, os
+
+is_linux = sys.platform.startswith('linux')
+
+if is_linux:
+    import SCons.Tool.gcc
+import SCons.Util
+import SCons.Warnings
+import SCons.Errors
+
+if not is_linux:
+    raise SCons.Errors.UserError("This tool doesn't work on non-Linux platforms")
+
+class OneapiError(SCons.Errors.InternalError):
+    pass
+
+class DetectCompiler(object):
+
+    def __init__(self):
+        root = '/opt/intel/oneapi/compiler/latest'
+        binp = os.path.join(root, 'linux', 'bin')
+        libp = os.path.join(root, 'linux', 'lib')
+        include = os.path.join(root, 'linux', 'include')
+        binarch = os.path.join(binp, 'intel64')
+        libarch = os.path.join(root, 'linux', 'compiler', 'lib', 'intel64_lin')
+        icx = os.path.join(binp, 'icx')
+        self.map = {}
+        sys.stdout.flush()
+        if not os.path.exists(root) or \
```

Comment:
(pylint-too-many-boolean-expressions) Too many boolean expressions in if statement (7/5)